sstables/sstable_set.cc

```diff
@@ -271,7 +297,10 @@ void partitioned_sstable_set::insert(shared_sstable sst) {
     _all->insert(sst);
     auto undo_all_insert = defer([&] () { _all->erase(sst); });
 
-    _all_runs[sst->run_identifier()].insert(sst);
+    // If sstable doesn't satisfy disjoint invariant, then place it in a new sstable run.
+    while (!_all_runs[sst->run_identifier()].insert(sst)) {
+        sst->generate_new_identifier();
```

Comment:
This should be generate_new_run_identifer() so we know what identifier is genetated.

Also, I think we should log a warning here.